gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/VersionPropsFileListener.java

```diff
@@ -62,24 +65,83 @@ public ChangeApplier prepareChange(List<? extends VFileEvent> events) {
                 .filter(Project::isInitialized)
                 .filter(Predicate.not(ComponentManager::isDisposed))
                 .filter(project -> versionPropsEvents.stream()
-                        .anyMatch(event -> event.getPath().startsWith(project.getBasePath())))
+                        .anyMatch(event -> event.getPath().startsWith(project.getBasePath())
+                                && !isFileMalformed(project, event.getFile())))
                 .toList();
 
         return new ChangeApplier() {
             @Override
             public void afterVfsChange() {
                 projectsAffected.forEach(project -> {
-                    ExternalSystemTaskExecutionSettings settings = new ExternalSystemTaskExecutionSettings();
-                    settings.setExternalProjectPath(project.getBasePath());
-                    settings.setTaskNames(Collections.singletonList("writeVersionsLock"));
-                    settings.setExternalSystemIdString(GradleConstants.SYSTEM_ID.toString());
-
-                    ExternalSystemUtil.runTask(
-                            settings, DefaultRunExecutor.EXECUTOR_ID, project, GradleConstants.SYSTEM_ID);
-                    ExternalSystemUtil.refreshProject(
-                            project.getBasePath(), new ImportSpecBuilder(project, GradleConstants.SYSTEM_ID).build());
+                    if (hasBuildSrc(project)) {
+                        runTaskThenRefresh(project);
+                    } else {
+                        refreshProjectWithTask(project);
+                    }
                 });
             }
         };
     }
+
+    private boolean hasBuildSrc(Project project) {
+        return Files.exists(Paths.get(project.getBasePath(), "buildSrc"));
+    }
+
+    private void runTaskThenRefresh(Project project) {
+        log.debug("Running task {} on project {}", VersionPropsFileListener.TASK_NAME, project.getName());
+        TaskCallback callback = new TaskCallback() {
+            @Override
+            public void onSuccess() {
+                log.debug("Task {} successfully executed", VersionPropsFileListener.TASK_NAME);
```

Comment:
TASK_NAME is sufficient, you don't need to prefix it with the class (same in the other places)
```suggestion
                log.debug("Task {} successfully executed", TASK_NAME);
```